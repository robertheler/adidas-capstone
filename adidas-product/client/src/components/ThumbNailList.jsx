import React, { Component } from 'react';
import styled from 'styled-components';
import ThumbNail from './ThumbNail.jsx';

const RenderDiv = styled.div`
  display: flex;
  flex-direction: row;
  position: relative;
  overflow: hidden;
  align-items: center;
  justify-content: center;
  height: 700px;
  z-index:5;
  -webkit-transition: all 0.3s;
  &:hover {
    -webkit-transform: scale(1.2);
  }
`;

class ThumbNailList extends Component {
  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  // Click handler is passed to child component ThumbNail
  // and will be called by child component
  handleClick(index, url) {
    this.props.handleClick(index);
  }

  render() {
    // Render an array of ThumbNails pics depending on
    // how many images there are for the current product
    let arrayOfThumbNails = [];
    for (let i = 0; i < this.props.images.length; i++) {
      // currently selected ThumbNail is passed prop isSelected = true
      // because it needs to be styled differnetly
      if(i === this.props.indexOfSelected) {
        arrayOfThumbNails.push(<ThumbNail key={i} index={i} image={this.props.images[i]} handleClick={this.handleClick} isSelected={true}/>)
      } else {
        arrayOfThumbNails.push(<ThumbNail key={i} index={i} image={this.props.images[i]} handleClick={this.handleClick} isSelected={false}/>)
      }
    }

    return (
      <RenderDiv>
        {arrayOfThumbNails}
      </RenderDiv>
    );
  }
}

export default ThumbNailList;