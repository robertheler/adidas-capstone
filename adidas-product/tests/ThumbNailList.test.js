import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ThumbNailList from '../client/src/components/ThumbNailList';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0],
    indexOfSelected: 0,
    handleClick: () => {}
  };
  const wrapper = shallow(<ThumbNailList images={props.product.colors[props.color].images} indexOfSelected={props.indexOfSelected} handleClick={props.handleClick}/>);
  return { wrapper, props };
}

describe('Title Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});