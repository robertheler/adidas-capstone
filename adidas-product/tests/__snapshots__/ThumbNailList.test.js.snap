// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Test Suite It should render correctly 1`] = `
<styled.div>
  <ThumbNail
    handleClick={[Function]}
    image="B42200_color0_pic0.jpg"
    index={0}
    isSelected={true}
    key="0"
  />
  <ThumbNail
    handleClick={[Function]}
    image="B42200_color0_pic2.jpg"
    index={1}
    isSelected={false}
    key="1"
  />
  <ThumbNail
    handleClick={[Function]}
    image="B42200_color0_pic7.jpg"
    index={2}
    isSelected={false}
    key="2"
  />
  <ThumbNail
    handleClick={[Function]}
    image="B42200_color0_pic8.jpg"
    index={3}
    isSelected={false}
    key="3"
  />
  <ThumbNail
    handleClick={[Function]}
    image="B42200_color0_pic9.jpg"
    index={4}
    isSelected={false}
    key="4"
  />
</styled.div>
`;
